src/Compilers/Core/Portable/Compilation/IStatement.cs

```diff
@@ -0,0 +1,461 @@
+        ImmutableArray<ICase> Cases { get; }
+    }
+
+    /// <summary>
+    /// Represents a C# case or VB Case statement.
+    /// </summary>
+    public interface ICase
+    {
+        /// <summary>
+        /// Clauses of the case. For C# there is one clause per case, but for VB there can be multiple.
+        /// </summary>
+        ImmutableArray<ICaseClause> Clauses { get; }
+        /// <summary>
+        /// Statements of the case.
+        /// </summary>
+        ImmutableArray<IStatement> Body { get; }
+    }
+
+    /// <summary>
+    /// Represents a clause of a C# case or a VB Case.
+    /// </summary>
+    public interface ICaseClause
+    {
+        /// <summary>
+        /// Kind of the clause.
+        /// </summary>
+        CaseKind CaseKind { get; }
+    }
+
+    /// <summary>
+    /// Kinds of cases.
+    /// </summary>
+    public enum CaseKind
+    {
+        /// <summary>
+        /// Indicates case x in C# or Case x in VB.
+        /// </summary>
+        SingleValue,
+        /// <summary>
+        /// Indicates Case Is op x in VB.
+        /// </summary>
+        Relational,
+        /// <summary>
+        /// Indicates Case x To Y in VB.
+        /// </summary>
+        Range,
+        /// <summary>
+        /// Indicates default: in C3 or Case Else in VB.
+        /// </summary>
+        Default
+    }
+
+    /// <summary>
+    /// Represents case x in C# or Case x in VB.
+    /// </summary>
+    public interface ISingleValueCaseClause : ICaseClause
+    {
+        /// <summary>
+        /// Case value.
+        /// </summary>
+        IExpression Value { get; }
+        /// <summary>
+        /// Relational operator used to compare the switch value with the case value.
+        /// </summary>
+        RelationalOperationKind Equality { get; }
+    }
+
+    /// <summary>
+    /// Represents Case Is op x in VB.
+    /// </summary>
+    public interface IRelationalCaseClause : ICaseClause
+    {
+        /// <summary>
+        /// Case value.
+        /// </summary>
+        IExpression Value { get; }
+        /// <summary>
+        /// Relational operator used to compare the switch value with the case value. 
+        /// </summary>
+        RelationalOperationKind Relation { get; }
+    }
+
+    // Represents Case x To y in VB.
+    public interface IRangeCaseClause : ICaseClause
+    {
+        /// <summary>
+        /// Minimum value of the case range.
+        /// </summary>
+        IExpression MinimumValue { get; }
+        /// <summary>
+        /// Maximum value of the case range.
+        /// </summary>
+        IExpression MaximumValue { get; }
+    }
+
+    /// <summary>
+    /// Represents an if statement in C# or an If statement in VB.
+    /// </summary>
+    public interface IIf : IStatement
+    {
+        /// <summary>
+        /// Clauses of the if. For C# there is one clause per if, but for VB there can be multiple.
+        /// </summary>
+        ImmutableArray<IIfClause> IfClauses { get; }
```

Comment:
Should be a single If condition and body with optional Else. (In both languages, "else if" is represented as an IfStatement in the Else.)